lib/client.js

```diff
@@ -48,6 +48,12 @@ function Client (broker, conn, req) {
 
   this.connDetails = null
 
+  // queue packets received before client fires 'connect' event. Prevents memory leaks on 'connect' event
+  this.parser._queue = []
+  this.parser.queueLimit = broker.queueLimit
```

Comment:
There is no need to copy this, we have `this.broker` where we can access this value from. I prefer to not add too many fields here because it will increase the memory footprint for every client.